controllers/pvc.go

```diff
@@ -1,9 +1,181 @@
 package controllers
 
 import (
+	"errors"
+	"fmt"
+
 	"github.com/go-logr/logr"
+	pvcv1alpha1 "github.com/konveyor/volume-snapshot-mover/api/v1alpha1"
+	snapv1 "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 func (r *DataMoverBackupReconciler) BindPVC(log logr.Logger) (bool, error) {
+	// Get datamoverbackup from cluster
+	dmb := pvcv1alpha1.DataMoverBackup{}
+	if err := r.Get(r.Context, r.NamespacedName, &dmb); err != nil {
+		return false, err
+	}
+	// Check if Volumesnapshot is present in the protected namespace
+	vs := snapv1.VolumeSnapshot{}
+	if err := r.Get(r.Context,
+		types.NamespacedName{Name: fmt.Sprintf("%s-volumesnapshot", dmb.Spec.VolumeSnapshotContent.Name), Namespace: r.NamespacedName.Namespace}, &vs); err != nil {
+		return false, errors.New("cloned volumesnapshot not available in the protected namespace")
+	}
+
+	// Create a PVC with the above volumesnapshot as the source
+	pvc := corev1.PersistentVolumeClaim{}
+	_ = r.Get(r.Context, types.NamespacedName{Name: fmt.Sprintf("%s-pvc", dmb.Spec.VolumeSnapshotContent.Name), Namespace: r.NamespacedName.Namespace}, &pvc)
```

Comment:
We need to check the error here right? Otherwise `pvc.Name == ""` will pass and we'll try to create a PVC when the Get call could have been a connection error.

Reply:
resolved. Thanks, Dylan